### PR TITLE
✨Separate CRDs in the Helm chart

### PR DESCRIPTION
**Pull request by francostellari**
## Summary

Move CRDs to the chart `crds/` folder to let Helm install them before applying the template

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
If we are going to comment out all usage of `config/crd` then I think it would be clearer to also delete `config/crd`.

**Comment by MikeSpreitzer** [comment]
This PR nullifies 
https://github.com/kubestellar/ocm-status-addon/blob/7521a376c0eb8961e12668f4c5f9f643ee3a8769/config/crd/kustomization.yaml#L7-L13 --- right? Is that deliberate? If so, why is this good?

I think that if we wanted to, we could continue applying `kustomize` to the CRD, but do it in separate commands. That is, remove `- ../crd` from `config/default/kustomization.yaml` and instead have the Makefile directly invoke `config/crd` in distinct command(s) that output the kustomized CRD YAML.

**Comment by francostellari** [comment]
@MikeSpreitzer I have pushed a commit to change the handling of the crds as you suggested.
I also rebased.
Please take another look when you have time.

File: Makefile [hunk 1]
```diff
@@ -176,27 +176,31 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
 .PHONY: deploy
 deploy: manifests kustomize ## Deploy manager to the K8s cluster specified in ~/.kube/config.
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	kubectl --context ${DEFAULT_IMBS_CONTEXT} apply -f config/crd/bases/control.kubestellar.io_workstatuses.yaml
```
Comment (MikeSpreitzer) [inline Makefile:179]: This is a problem because it uses the CRDs without kustomization.

File: Makefile [hunk 2]
```diff
@@ -176,27 +176,31 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
...
 	$(KUSTOMIZE) build config/default | kubectl --context ${DEFAULT_IMBS_CONTEXT} apply -f -
 
 .PHONY: undeploy
 undeploy: ## Undeploy manager from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
 	$(KUSTOMIZE) build config/default | kubectl --context ${DEFAULT_IMBS_CONTEXT} delete --ignore-not-found=$(ignore-not-found) -f -
+	kubectl --context ${DEFAULT_IMBS_CONTEXT} delete --ignore-not-found=$(ignore-not-found) -f config/crd/bases/control.kubestellar.io_workstatuses.yaml
```
Comment (MikeSpreitzer) [inline Makefile:185]: Also using the CRDs without kustomization

File: Makefile [hunk 3]
```diff
@@ -176,27 +176,31 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified
...
 
 .PHONY: chart
 chart: manifests kustomize
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(shell echo ${IMG} | sed 's/\(:.*\)v/\1/')
 	$(KUSTOMIZE) build config/default > chart/templates/operator.yaml
+	@mkdir -p chart/crds
+	$(KUSTOMIZE) build config/crd > chart/crds/crds.yaml
```
Comment (MikeSpreitzer) [inline Makefile:192]: This is a problem, see https://github.com/kubestellar/ocm-status-addon/blob/7521a376c0eb8961e12668f4c5f9f643ee3a8769/config/crd/kustomization.yaml#L1-L2

**Review by MikeSpreitzer — COMMENTED** [review]
/lgtm

File: Makefile (hunk 1 above)
Comment (MikeSpreitzer) [reply Makefile]: Fixed in later rev.

File: Makefile (hunk 2 above)
Comment (MikeSpreitzer) [reply Makefile]: Fixed in later rev.

File: Makefile (hunk 3 above)
Comment (MikeSpreitzer) [reply Makefile]: Done.

**Review by MikeSpreitzer — COMMENTED** [review]
/lgtm

**Review by pdettori — APPROVED** [review]
/lgtm

**Pull request merged**